File: backReact/src/main/java/TP3/backReact/entities/PedidoDetalle.java
package TP3.backReact.entities;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.io.Serializable;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class PedidoDetalle implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private int cantidad;

    @ManyToOne(cascade = CascadeType.PERSIST)
    @JoinColumn(name = "id_instrumento")
    private Instrumento instrumento;

    @ManyToOne(cascade = CascadeType.PERSIST)
    @JoinColumn(name="id_pedido")
    @JsonBackReference
    private Pedido pedido;

}

File: Tp3_React_Incial/src/Entities/PedidoDetalle.ts
import Instrumento from "./Intrumento";
import Pedido from "./Pedido";

export default class PedidoDetalle{
    id:number = 0;
    cantidad:number = 0;
    pedido:Pedido = new Pedido();
    instrumento:Instrumento = new Instrumento();
}